modules/table/src/main/java/org/apache/ignite/internal/table/distributed/raft/snapshot/incoming/IncomingSnapshotCopier.java

```diff
@@ -195,8 +195,8 @@ private CompletableFuture<?> prepareMvPartitionStorageForRebalance(Executor exec
             return completedFuture(null);
         }
 
-        return CompletableFuture.supplyAsync(() -> partitionSnapshotStorage.partition().reCreateMvPartitionStorage(), executor)
-                .thenCompose(mvPartitionStorage -> {
+        return partitionSnapshotStorage.partition().reCreateMvPartitionStorage()
```

Comment:
Looks like MV partition will be destroyed on the current thread. Before this change, it would be executed on `executor`. Is this ok? Can partition destroy process be heavy?

Reply:
Cool!